src/gui/ar/meshPath.js

```diff
@@ -351,15 +391,16 @@ export class MeshPath extends THREE.Group
         let brightness = this.wallBrightness;
 
         pointIndicesThatNeedUpdate.forEach(index => {
-            let bufferIndices = this.getBufferIndices(index);
-            bufferIndices.forEach(bfrIndex => {
+            let colorBufferIndices = this.getBufferIndices(index, COMPONENTS_PER_COLOR);
+            colorBufferIndices.forEach(bfrIndex => {
                 let newColor = {
                     r: this.currentPoints[index].color[0],
                     g: this.currentPoints[index].color[1],
-                    b: this.currentPoints[index].color[2]
+                    b: this.currentPoints[index].color[2],
+                    a: this.currentPoints[index].length === 3 ? 255 : this.currentPoints[index].color[3]
```

Comment:
I believe this should be currentPoints[index].color.length

Reply:
Good catch!